D3D12Bundles/D3D12Bundles.cs

```diff
@@ -237,79 +225,97 @@ struct Vertex {
+                        DenyList = new Vortice.DXGI.Debug.InfoQueueFilterDescription {
+                            Ids = hide
+                        }
+                    };
+                    dxgiInfoQueue.AddStorageFilterEntries(DXGI.DebugDxgi, filter);
+                    dxgiInfoQueue.Dispose();
+                }
             }
 #endif
 
-            DXGI.CreateDXGIFactory2(dxgiFactoryDebugMode, out IDXGIFactory4? factory);
+            DXGI.CreateDXGIFactory2(dxgiFactoryDebugMode, out IDXGIFactory4? factory).CheckError();
 
+            ID3D12Device device;
             if (mUseWarpDevice) {
                 Result warpResult = factory!.EnumWarpAdapter(out IDXGIAdapter? warpAdapter);
                 if (warpResult.Failure)
                     throw new COMException("EnumWarpAdaptor creation failed", warpResult.Code);
 
-                mDevice = D3D12.D3D12CreateDevice<ID3D12Device>(warpAdapter, Vortice.Direct3D.FeatureLevel.Level_11_0);
+                device = D3D12.D3D12CreateDevice<ID3D12Device>(warpAdapter, Vortice.Direct3D.FeatureLevel.Level_11_0);
             }
             else {
                 IDXGIAdapter1? hardwareAdapter = null;
                 //We could pull this from https://github.com/microsoft/DirectX-Graphics-Samples/blob/3e8b39eba5facbaa3cd26d4196452987ac34499d/Samples/UWP/D3D12HelloWorld/src/HelloTriangle/DXSample.cpp#L43
                 //But for now, leave it up to Vortice to figure out...
                 //GetHardwareAdapter(factory.Get(), &hardwareAdapter);  
 
-                mDevice = D3D12.D3D12CreateDevice<ID3D12Device>(hardwareAdapter, Vortice.Direct3D.FeatureLevel.Level_11_0);
+                device = D3D12.D3D12CreateDevice<ID3D12Device>(hardwareAdapter, Vortice.Direct3D.FeatureLevel.Level_11_0);
             }
 
-            // Describe and create the command queue.
-            mCommandQueue = mDevice.CreateCommandQueue(new CommandQueueDescription(CommandListType.Direct));
-            mCommandQueue.Name = nameof(mCommandQueue);
-
-            // Describe and create the swap chain.
-            var swapChainDesc = new SwapChainDescription1 {
-                BufferCount = FrameCount,
-                Width = Width,
-                Height = Height,
-                Format = Format.R8G8B8A8_UNorm,
-                BufferUsage = Usage.RenderTargetOutput,
-                SwapEffect = SwapEffect.FlipDiscard,
-                SampleDescription = SampleDescription.Default,
-            };
-
-            // Swap chain needs the queue so that it can force a flush on it.
-            using IDXGISwapChain1 swapChain = factory!.CreateSwapChainForHwnd(mCommandQueue, base.Handle, swapChainDesc);
-            mSwapChain = swapChain.QueryInterface<IDXGISwapChain3>();
-            mFrameIndex = mSwapChain.CurrentBackBufferIndex;
-
-            // Create descriptor heaps.
+#if DEBUG
+            // Configure debug device (if active).
```

Comment:
More debugging code setup copied from D3D12Mutiny.  Definitely has a performance impact, so use release builds to avoid that.